dimension/dimension_test.go

```diff
@@ -37,19 +72,29 @@ func TestAddNodeIDToDimensionReturnsOK(t *testing.T) {
 			},
 		}
 
-		dimension := &dimension.Store{Storer: mockedDataStore}
-		dimension.AddNodeID(w, r)
+		auditorMock := newAuditorMock()
+
+		datasetAPI := getAPIWithMockedDatastore(mockedDataStore, &mocks.DownloadsGeneratorMock{}, auditorMock, &mocks.ObservationStoreMock{})
+		datasetAPI.Router.ServeHTTP(w, r)
 
 		So(w.Code, ShouldEqual, http.StatusOK)
-		So(len(mockedDataStore.GetInstanceCalls()), ShouldEqual, 1)
+		So(len(mockedDataStore.GetInstanceCalls()), ShouldEqual, 2)
```

Comment:
is this right? doesn't seem right that we would have to call getInstance twice

Reply:
Yes it is right, GetInstance func is called twice due to check wrapper around this route; it checks the instance is not published before entering handler, if it is published it throws 403 (forbidden error).